sdk/scheduler/src/main/java/com/mesosphere/sdk/config/validate/UserCannotChange.java

```diff
@@ -0,0 +1,98 @@
+package com.mesosphere.sdk.config.validate;
+
+import com.mesosphere.sdk.specification.PodSpec;
+import com.mesosphere.sdk.specification.ServiceSpec;
+
+import java.util.*;
+
+/**
+ * Configuration validator which validates that a {@link com.mesosphere.sdk.specification.ServiceSpec}'s user
+ * cannot change between deployments.
+ *
+ * This is to prevent change in user-level permissions while accessing files across updates. If a
+ * less-privileged user replaces a more-privilege user (eg. "root" -> "nobody"), this is a breaking change
+ * as "nobody" will not have access to "root"-owned files in persistent volumes.
+ */
+public class UserCannotChange implements ConfigValidator<ServiceSpec> {
+    private static final String USER_CHANGED_ERROR_MESSAGE =
+            "User for old pod type %s must remain the same across deployments. Expected: '%s', given: '%s'";
+
+    @Override
+    public Collection<ConfigValidationError> validate(Optional<ServiceSpec> oldConfig, ServiceSpec newConfig) {
+        if (!oldConfig.isPresent()) {
+            return Collections.emptyList();
+        }
+
+        List<ConfigValidationError> errors = new ArrayList<>();
+
+        // The corresponding old and new pods must have the same user.
+        Iterator<PodSpec> oldPods = oldConfig.get().getPods().iterator();
+        Iterator<PodSpec> newPods = newConfig.getPods().iterator();
+        PodSpec oldPod = null, newPod;
+        String oldUser = "", newUser;
+
+        while (oldPods.hasNext() && newPods.hasNext()) {
```

Comment:
I don't think you should rely on oldPods and newPods being in the same order.  Consider in particular the addition of a new Pod at the beginning of the list.  This would throw things off.  Probably need to use Maps.